src/utils/fetchNftsWithDuration.ts

```diff
@@ -0,0 +1,63 @@
+/*-
+ *
+ * Hedera Token Holders List
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import { BalancesWithNFT } from '@/types/balances-response';
+import { nodeUrl } from '@/utils/const';
+import { NftsResponseType } from '@/types/nfts-response';
+import { getHolderAndDuration } from 'nft-utilities/src/getHolderAndDuration/index';
+import { changeDurationToDate } from '@/utils/changeDurationToDate';
+
+export const fetchNftsWithDuration = async (nftBalances: BalancesWithNFT[]): Promise<BalancesWithNFT[]> => {
+  for (const nftBalance of nftBalances) {
+    // Skip if the NFT doesn't have a duration. Duration is not required
+    if (!nftBalance.duration) continue;
+
+    let countSuccessfullyNfts = 0;
```

Comment:
let's name it `nftsMeetingDurationCondition`

Reply:
Done